File: Day324/BTInorderTraversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func inorder(root *TreeNode, answer *[]int) {
    if root == nil {
        return 
    }
    inorder(root.Left, answer)
    *answer = append(*answer, root.Val)
    inorder(root.Right, answer)
}
func inorderTraversal(root *TreeNode) []int {
    answer := make([]int , 0)
    inorder(root, &answer)
    return answer
}

File: Day324/BTInorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> answer;
    public void inorder(TreeNode root) {
        if (root == null) {
            return;
        }
        inorder(root.left);
        answer.add(root.val);
        
        inorder(root.right);
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        answer = new ArrayList<>();
        inorder(root);
        return answer;
    }
}